### DISCUSSION TITLE
Using section 3 of Lab 2.04 as the Unit project instead of Zork

### DISCUSSION THREAD

**Discussion by jdonwells** [TEALSK12/2nd-semester-introduction-to-computer-science#339] [Ideas]
I feel like Zork is too much for a final project to Unit 2. We have to insert lesson 2.06 in order to make it work out. Even then we end up with an if else statement that is 70 or more lines long. Let's not teach them that is okay to do. I want to save the Zork project until Unit 6 when they have dictionaries to do translations.

Instead, let's take the create a quiz exercise from lab 2.04 and make it more important. Let's build it up in terms of Artificial Intelligence on a small simple scale. Let them choose an area of expertise and create an expert system based on that knowledge.

Here is my slide show introduction so far. https://docs.google.com/presentation/d/1V8mrEX4i5Nk6THf1ucCpkOvOLMLIGTeo9nzf6HxSwxU/edit?usp=sharing

# Unit 2 Project - Expert System

## 1. Pay Attention

If you are reading this instead of watching the lesson stop. The lesson will contain important hints about how to structure your program.

## 2. Design a simple Expert System that uses a single layer neural network

You will need to:

* Choose a subject you are an expert at.
* Choose a single decision that needs to be made in your subject.
* Choose important questions that must be asked to give expert level advice on that decision.

## 3. Build your system

Create an expert system using a list as the hidden layer of your neural network. Use the input() function to ask questions. Raise and lower the score with an arithmetic expression. After answering all the questions, find the maximum score and print the corresponding decision with the output() function. Watch the lesson carefully to see how you can use what you already know to make an expert system work.

## 4. Requirements

* 6 or more output choices
* 6 or more questions
* Questions should not directly ask if you want a specific choice. They must show expertise.
* At least 3 questions should be if: else: and not just if:.
* Choose any subject you want. The greater your expertise the better the result will be.
* Use blank lines to make your code easier to read. Use comments as needed.

**Comment by RandyCHS** [comment]
@jdonwells I'm intrigued by this idea for the unit 2 project. This is my 4th year teaching this curriculum and I've always found that many students struggle with all the cascading conditionals required for the canonical solution to the Text Monster problem.  My students just started unit 2 last week and will be ready to start this project next week. Might you have a worked sample solution for this problem?

**Comment by jdonwells** [comment]
I take them through a full example but with only 3 choices. It is part of the slide show. They are then given full access to that slide show after the first 2 days of work. Since I give them the structure in the lesson I gave them 3 days to work on this. 

The slide show is liked above. repl.it https://replit.com/@jdonwells/206-project#main.py

```
food = ['pancakes', 'bacon', 'eggs']
score = [0, 0, 0]

PANCAKES = 0
BACON = 1
EGGS = 2

print('Please answer each question with "y" for yes and "n" for no.')

user_input = input('Are you a vegetarian? ')
if user_input == 'y':
    score[PANCAKES] = score[PANCAKES] + 3
    score[BACON] = score[BACON] - 5
    score[EGGS] = score[EGGS] - 5

user_input = input('Are you watching your sodium? ')
if user_input == 'y':
    score[PANCAKES] = score[PANCAKES] + 1
    score[BACON] = score[BACON] - 5

user_input = input('Are you diabetic? ')
if user_input == 'y':
    score[PANCAKES] = score[PANCAKES] - 1
    score[BACON] = score[BACON] + 3
    score[EGGS] = score[EGGS] + 3

user_input = input('Do you like maple syrup? ')
if user_input == 'y':
    score[PANCAKES] = score[PANCAKES] + 1
else:
    score[PANCAKES] = score[PANCAKES] - 2

user_input = input('Do you like butter? ')
if user_input == 'y':
    score[PANCAKES] = score[PANCAKES] + 1

best_choice = 0
if score[best_choice] < score[1]:
    best_choice = 1
if score[best_choice] < score[2]:
    best_choice = 2

print("Your best choice will be " + food[best_choice] + ".")
```
